ext/hash/php_hash_sha.h

```diff
@@ -56,10 +56,10 @@ PHP_HASH_API void PHP_SHA256Update(PHP_SHA256_CTX *, const unsigned char *, size
 void SHA256_Transform_sse2(uint32_t state[PHP_STATIC_RESTRICT 8], const uint8_t block[PHP_STATIC_RESTRICT 64], uint32_t W[PHP_STATIC_RESTRICT 64], uint32_t S[PHP_STATIC_RESTRICT 8]);
 #endif
 
-#if (defined(__i386__) || defined(__x86_64__)) && defined(HAVE_IMMINTRIN_H)
+#if ((defined(__i386__) || defined(__x86_64__)) && defined(HAVE_IMMINTRIN_H)) || ((defined(_M_X64) || defined(_M_IX86)) && defined(ZEND_INTRIN_SSSE3_RESOLVER))
```

Comment:
Should this also check for HAVE_IMMINTRIN_H on Windows? The header file in unconditionally included there. Or is that what the `ZEND_INTRIN_SSSE3_RESOLVER` is for?

Reply:
`HAVE_IMMINTRIN_H` is never defined on Windows (not only for MSVC), but it is assumed that the file is always available. This might be something to clean up.

And actually, I don't think that `ZEND_INTRIN_SSSE3_RESOLVER` is needed here; the macro is defined if `(defined(ZEND_WIN32) && (!defined(_M_ARM64)` which should be the same as `(defined(_M_X64) || defined(_M_IX86)` for MSVC. And somehow that definition looks not quite right regarding clang (don't think that icc is still supported there, but not sure; never tested) on Windows (the usual confusion between WIN32 and MSVC, a trap into which I fell quite often):

https://github.com/php/php-src/blob/42497c1ea56a8452af84ea9c2177c1175f4136d6/Zend/zend_portability.h#L556-L559

Reply:
Agreed that `ZEND_INTRIN_SSSE3_RESOLVER` is not needed here